src/benchmarklib/table_builder.hpp

```diff
@@ -0,0 +1,240 @@
+  using value_type = T;
+};
+
+template <typename T>
+struct GetValueType<T, std::enable_if_t<is_optional_v<T>>> {
+  using value_type = typename T::value_type;
+};
+
+// get the value_type (lets call it V) of std::optional<V> and V using the same syntax - needed for templating
+template <typename T>
+using get_value_type = typename GetValueType<T>::value_type;
+
+// check if std::optional<V> or V is null using the same syntax - needed for templating
+template <typename T>
+constexpr bool is_null(const T& maybe_optional) {
+  if constexpr (is_optional_v<T>) {
+    return !maybe_optional.has_value();
+  } else {
+    return false;
+  }
+}
+
+// get the value of std::optional<V> or V using the same syntax - needed for templating
+template <typename T>
+get_value_type<T>& get_value(T& maybe_optional) {
+  if constexpr (is_optional_v<T>) {
+    return maybe_optional.value();
+  } else {
+    return maybe_optional;
+  }
+}
+
+}  // namespace
+
+namespace opossum {
+
+/**
+ * Helper to build a table with a static column layout, specified by ctor args types and names. Keeps a
+ * data vector for each column and appends values to them in append_row(). For nullable columns an additional
+ * is_null_vector is kept. Automatically creates chunks in accordance with the specified chunk size.
+ */
+template <typename... DataTypes>
+class TableBuilder {
+ public:
+  // names is a list of strings defining column names, types is a list of equal length defining respective column types
+  // types may contain std::optional<?>, which will result in a nullable column, otherwise columns are not nullable
+  template <typename... Names>
+  TableBuilder(size_t chunk_size, const hana::tuple<DataTypes...>& types, const hana::tuple<Names...>& names,
+               opossum::UseMvcc use_mvcc, size_t estimated_rows = 0)
+      : _use_mvcc(use_mvcc), _estimated_rows_per_chunk(std::min(estimated_rows, chunk_size)) {
+    BOOST_HANA_CONSTANT_ASSERT(hana::size(names) == hana::size(types));
+
+    // Iterate over the column types/names and create the columns.
+    auto column_definitions = TableColumnDefinitions{};
+    hana::fold_left(hana::zip(names, types), column_definitions,
+                    [](auto& definitions, const auto& name_and_type) -> decltype(definitions) {
+                      auto name = name_and_type[hana::llong_c<0>];
+                      auto type = name_and_type[hana::llong_c<1>];
+
+                      auto data_type = data_type_from_type<get_value_type<decltype(type)>>();
+                      auto is_nullable = is_optional<decltype(type)>();
+
+                      definitions.emplace_back(name, data_type, is_nullable);
+
+                      return definitions;
+                    });
+    _table = std::make_shared<Table>(column_definitions, TableType::Data, chunk_size, use_mvcc);
+
+    // Reserve some space in the vectors
+    hana::for_each(_data_vectors, [&](auto&& vector) { vector.reserve(_estimated_rows_per_chunk); });
```

Comment:
I think everywhere else (e.g. ValueSegment) we have established to call these vector `values` and `null_values`